tests/transforms/test_split_non_commuting.py

```diff
@@ -122,765 +119,580 @@ def complex_wires_processing_fn(results):
+        assert qml.math.allclose(fn([0.1]), [0.1, 0.05])
+
+    def test_mix_measurement_types_qwc(self):
+        """Tests multiple measurement types can be handled by qwc grouping"""
+
+        initial_tape = qml.tape.QuantumScript(
+            [qml.X(0)],
+            measurements=[
+                # [Z(0), X(0), Z(0), X(0), Y(1), Z(1), Z(0) @ Z(1), Z(1), Z(0)]
+                qml.expval(qml.Z(0)),
+                qml.expval(qml.Z(0) + qml.X(0)),
+                qml.var(qml.Z(0)),
+                qml.probs(op=qml.X(0)),
+                qml.counts(qml.Y(1)),
+                qml.sample(qml.Z(1)),
+                qml.probs(wires=[0, 1]),
+                qml.sample(wires=[1]),
+                qml.counts(wires=[0]),
             ],
-        ],
-    )
-    def test_tape_with_non_pauli_obs(self, non_pauli_obs):
-        """Tests that the tape is split correctly when containing non-Pauli observables"""
-
-        obs_list = single_term_obs_list + non_pauli_obs
+        )
 
-        measurements = [
-            qml.expval(c * o) for c, o in zip([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7], obs_list)
+        # ((0, 2, 5, 6, 7, 8), (1, 3, 4))
+        tapes, fn = split_non_commuting(initial_tape, grouping_strategy="qwc")
+        assert len(tapes) == 2
+        assert tapes[0].measurements == [
+            qml.expval(qml.Z(0)),
+            qml.var(qml.Z(0)),
+            qml.sample(qml.Z(1)),
+            qml.probs(wires=[0, 1]),
+            qml.sample(wires=[1]),
+            qml.counts(wires=[0]),
         ]
-        tape = qml.tape.QuantumScript([], measurements, shots=100)
-
-        expected_tapes_no_grouping = [
-            qml.tape.QuantumScript([], [qml.expval(o)], shots=100) for o in obs_list
+        assert tapes[1].measurements == [
+            qml.expval(qml.X(0)),
+            qml.probs(op=qml.X(0)),
+            qml.counts(qml.Y(1)),
         ]
-
-        tapes, fn = split_non_commuting(tape, grouping_strategy=None)
-        for actual_tape, expected_tape in zip(tapes, expected_tapes_no_grouping):
-            qml.assert_equal(actual_tape, expected_tape)
-        assert qml.math.allclose(
-            fn([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7]), [0.01, 0.04, 0.09, 0.16, 0.25, 0.36, 0.49]
+        results = fn(
+            [
+                [
+                    0.1,
+                    0.2,
+                    np.array([1.0, 1.0]),
+                    np.array([1.0, 0.0, 0.0, 0.0]),
+                    np.array([0, 0]),
+                    {"0": 2},
+                ],
+                [0.5, np.array([0.5, 0.5]), {"1": 2}],
+            ]
+        )
+        expected = (
+            0.1,
+            0.1 + 0.5,
+            0.2,
+            np.array([0.5, 0.5]),
+            {"1": 2},
+            np.array([1.0, 1.0]),
+            np.array([1.0, 0.0, 0.0, 0.0]),
+            np.array([0, 0]),
+            {"0": 2},
+        )
+        assert len(results) == len(expected)
+        for res, exp in zip(results, expected):
```

Comment:
I'd generally discourage shadowing names from math operations --- `exp` would be best named something else